specificity.go

```diff
@@ -0,0 +1,25 @@
+package cascadia
+
+// Specificity is the CSS specificity as defined in
+// https://www.w3.org/TR/selectors/#specificity-rules
+// with the convention Specificity = [A,B,C].
+type Specificity [3]int
+
+// returns `true` if s < other (strictly), false otherwise
+func (s Specificity) Less(other Specificity) bool {
+	for i := range s {
+		if s[i] < other[i] {
+			return true
+		}
+		if s[i] > other[i] {
+			return false
+		}
+	}
+	return false
+}
+
+func (s *Specificity) Add(other Specificity) {
```

Comment:
I think it would be better for Add to be a value method, instead of mutating the reciever:

```go
func (s Specificity) Add(other Specificity) Specificity
```

Reply:
Yeah, it's more like a + operator now